packages/graphql-transformer-core/src/util/amplifyUtils.ts

```diff
@@ -145,7 +145,7 @@ async function ensureMissingStackMappings(config: ProjectOptions) {
               missingStackMappings[resourceId] = stackName;
             }
           }
-          const outputIdsInStack = Object.keys(lastDeployedStack.Outputs);
+          const outputIdsInStack = Object.keys(lastDeployedStack.Outputs || {});
```

Comment:
Since outputs are optional it is possible the previous stack may have not outputs.
If possible could the same check be added to the root stack outputs as well?
- [L165](https://github.com/aws-amplify/amplify-cli/blob/4dc4d063a451ca783a6bcea6748f277d390d262e/packages/graphql-transformer-core/src/util/amplifyUtils.ts#L165)

Reply:
Sure done!